src/run-on-arch.js

```diff
@@ -126,7 +126,7 @@ async function main() {
   console.log('Configuring Docker for multi-architecture support');
   await exec(
     path.join(__dirname, 'run-on-arch.sh'),
-    [ dockerFile, containerName, ...dockerRunArgs ],
+    [ arch, dockerFile, containerName, ...dockerRunArgs ],
```

Comment:
Move `arch` to be the third argument. This will reduce the number of changes in the .sh file